frontend/src/components/StockData/StockData.jsx

```diff
@@ -16,20 +17,28 @@ export default function StockData(props) {
                 setStockNotFound(false);
 
                 const databaseResponse = await axios.get(
-                    `http://localhost:3000/stocks/${ticker}`
+                    `http://localhost:3000/stocks/${ticker}`,
+                    { withCredentials: true }
                 );
-                const stockExistsInDatabase = databaseResponse.data
+
+                const stockExistsInDatabase = databaseResponse.data.stock
```

Comment:
stock is not a boolean. Instead of `databaseResponse.data.stock ?` it should be an explicit:
`databaseResponse.data.stock !== undefined && databaseResponse.data.stock !== null`
Or the shorter form:
`databaseResponse.data.stock != null`